### ISSUE TITLE
[CI] Linting of docs

### ISSUE THREAD

**Issue by eddiebergman** [openml/openml-python#1296]
Currently there is no linter for doc and docstrings.

As a follow up, this should really be linted. I would advocate using [`ruff`](https://github.com/astral-sh/ruff) but theoretically that's blocked on still supporting python 3.6 (#1083).

Should work if working in an environment that's 3.7+ though.

The alternative is to use [`pydocstyle`](http://www.pydocstyle.org/en/stable/).

This would prevent issues like #716 from appearing. (which I guess can be closed now?)

**Comment by PGijsbers** [comment]
`ruff` is the way to go here I think. Since we want to drop 3.6 and fewer tools for the same functional behavior is better.

**Comment by ravjot07** [comment]
I would like to work on this. Is this issue still open?

**Comment by Omswastik-11** [comment]
Hi @PGijsbers and @eddiebergman !! 👋
I want to contribute to this issue is it open ?

**Comment by PGijsbers** [comment]
Yes, we would like to have the [ruff linter](https://docs.astral.sh/ruff/linter/) also be part of our pre-commit hooks.
However, we do have a large influx of PRs right now, so while you can go ahead and add the hook and make sure it works correctly, please abstain from committing fixes for linting errors right now. We can take up the PR once others have merged. Otherwise it will lead to a large amount of merge conflicts.

**Comment by Omswastik-11** [comment]
Ok Thanks . I will look into it after PR got merged

**Comment by ravjot07** [comment]
Ya this is good idea we will work on this after these PRs got merged
